wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet.hs

```diff
@@ -485,13 +508,19 @@ data UnknownHdAccount =
 
     -- | Unknown account (implies the root is known)
   | UnknownHdAccount HdAccountId
-  deriving Eq
+  deriving (Generic, Eq, Show)
+
 
 instance Arbitrary UnknownHdAccount where
     arbitrary = oneof [ UnknownHdAccountRoot <$> arbitrary
                       , UnknownHdAccount <$> arbitrary
                       ]
 
+deriving newtype instance ToJSON (V1 UnknownHdAccount)
+deriving newtype instance FromJSON (V1 UnknownHdAccount)
+instance ToJSON UnknownHdAccount
+instance FromJSON UnknownHdAccount
+
```

Comment:
Ah, yes, good question, @ouroboros spotted the inconsistency as well.

There is a rationale behind it, though. We decided to do this with @edsko a while ago to signal that this is an error reporting behaviour specific to `V1`, which might change in the future (for example if we plug external wallets or non-hd wallets in the future). It would allow us to do the same trick of deriving particular instances for `V1 Foo` rather than just `Foo`, which might be handy.

But you are right, we should do this consistently across the board. That's an oversight, oops 😉